_source/lib/apollos.coffee

```diff
@@ -81,41 +81,75 @@ Apollos.person = (user) ->
     person = Apollos.people.findOne
       guid: new RegExp(user.personGuid, "i")
 
-
-  # we shouldnt have any code for rock in the core package
+  # TODO we shouldnt have any code for rock in the core package
   # this will probably need to adjust the rock package
 
-  # if not person and user.rock and user.rock.personId
-  #   person = Apollos.people.findOne
-  #     personId: user.rock.personId
+  # PersonGuid would fall into the same category.  The only difference is that
+  # it doesn't live under the rock object, though it should.
+  # In fact, all relationships are made through Rock Ids and guids, which means
+  # all relationships need to be resolved within apollos-rock
+  if not person and user.rock?.personId
+    person = Apollos.people.findOne
+      personId: user.rock.personId
 
   return person or {}
 
-
-
-
-
-
 if Meteor.server
   Accounts.onCreateUser (options, user) ->
 
+    user.updatedBy or= Apollos.name
+
     if options.profile
       user.profile = options.profile
 
     if user.profile?.guest is true
       return user
 
-    person = Apollos.person user
-
-    # # no existing user so create one
-    if not Object.keys(person).length
-      if not user.personGuid
-        user.personGuid = Apollos.utilities.makeNewGuid()
-
-      Apollos.people.upsert({guid: user.personGuid}, {
-        $set:
-          preferredEmail: user.emails[0].address
-      })
+    email = user.emails[0].address
+
+    Meteor.setTimeout ->
```

Comment:
Database calls are now async.  There is an Apollos.user.onCreate callback now that works like Meteor.startup.  Timeout is gone.